app/src/main/java/rick_and_morty/ui/character_details/CharacterDetailsViewModel.kt

```diff
@@ -0,0 +1,46 @@
+package rick_and_morty.ui.character_details
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+import rick_and_morty.data.repository.CharacterRepository
+import javax.inject.Inject
+
+@HiltViewModel
+class CharacterDetailsViewModel @Inject constructor(
+    private val characterRepository: CharacterRepository,
+    savedStateHandle: SavedStateHandle
+) : ViewModel(){
+
+
+    private var _characterDetails = MutableStateFlow(CharacterDetailsUiState())
```

Comment:
we use `val` by default. Read only data is easier to reason about. We use `var` very rarely, typically for performance reasons.

Reply:
done